apps/ios/Shared/Model/ChatModel.swift

```diff
@@ -125,6 +125,35 @@ final class ChatModel: ObservableObject {
     @Published var pasteboardHasStrings: Bool = UIPasteboard.general.hasStrings
     @Published var networkInfo = UserNetworkInfo(networkType: .other, online: true)
 
+    private var chatMoveSubject = PassthroughSubject<Void, Never>()
+    private var bag = Set<AnyCancellable>()
+    private var chatIDsToMove = [Chat.ID]() {
+        didSet {
+            if !chatIDsToMove.isEmpty { chatMoveSubject.send() }
+        }
+    }
+
+    private func moveChatIDs() {
+        withAnimation(chatId == nil ? .default : nil) {
+            chats.move(
+                fromOffsets: IndexSet(
+                    self.chatIDsToMove.compactMap {
+                        chatId in self.chats.firstIndex(where: { $0.id == chatId })
+                    }
+                ),
+                toOffset: .zero
+            )
+            chatIDsToMove = [Chat.ID]()
+        }
+    }
+
+    init() {
+        chatMoveSubject
+            .throttle(for: 10, scheduler: DispatchQueue.main, latest: true)
+            .sink { self.moveChatIDs() }
+            .store(in: &bag)
+    }
+
```

Comment:
# Explanation
1. Modifying `chatIDsToMove` will send a "ping" to `chatMoveSubject`
2. Subject will call `moveChatIDs()` on first event and then ignore any upcoming events.
3. Subject fill fire after time is up to resolve the state